### PR TITLE
Feature: Sign in Anonymously with Ethereum

### PR DESCRIPTION
**Pull request by arnonrdp**
- Log in anonymously with Ethereum;
- Add an Ethereum account to a current account;
- Add wallets to your logged-in account.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,12 +23,14 @@
...
     "dotenv": "^16.3.1",
     "echarts": "^5.4.3",
+    "ethers": "^6.7.1",
     "firebase": "^10.4.0",
     "pinia": "^2.1.6",
     "pinia-plugin-persistedstate": "^3.2.0",
     "quasar": "^2.12.7",
     "secure-ls": "^1.2.6",
     "sha1": "^1.1.1",
+    "siwe": "^2.1.4",
```
Comment (stravid87) [inline package.json:33]: Excellent. Double checked, this is the official version.

File: src/stores/auth.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      try {
+        const signer = await provider.getSigner()
+
+        this._address = signer.address
+
+        this._message = await this.createSiweMessage(this._address)
+
+        this._signature = await signer.signMessage(this._message)
+
+        Notify.create({ message: 'Successfully signed in with Ethereum', type: 'positive' })
```
Comment (stravid87) [inline src/stores/auth.js:85]: Nope. Unfortunately not yet! You've celebrated too soon! You now need to send to a custom API back end. You may have this on We've got poems but not exactly here. Was that intentional?

**Review by stravid87 — CHANGES_REQUESTED** [review]
Did you specifically leave out the API call to a back end? This is necessary to complete the process. This is really just the first half.

File: src/stores/auth.js (hunk 2 above)
Comment (arnonrdp) [reply src/stores/auth.js:85]: I think it is possible to do without the dedicated Backend, or am I wrong?

**Review by hermannleboss — COMMENTED** [review]
For more security we need a backend for SIWE, 
But if we have only the backend for SIWE, Is it really useful?

For my implementation on crypto Tips, i turn all the backend (firebase) into a real backend (node + postgres)

Now the question is what should be the usage ?

**Comment by arnonrdp** [comment]
@hermannleboss @stravid87 @NJavokhir 

> ## Implementing "Sign In With Ethereum" (SIWE) Without a Dedicated Backend
> 
> Implementing **Sign In With Ethereum** (SIWE) typically involves interacting with the Ethereum blockchain, which is a decentralized and distributed network. While it's technically possible to perform some Ethereum-related operations directly on the client side (such as signing transactions with a private key), it's generally not recommended for security reasons.
> 
> Here are some considerations for implementing SIWE without a dedicated backend:
> 
> 1. **Security**: Handling private keys or sensitive Ethereum-related operations on the client side can expose your application to security risks. Storing private keys on the client can make them vulnerable to theft or misuse. A dedicated backend can provide a more secure environment for handling such sensitive data.
> 
> 2. **Authorization and Access Control**: To ensure proper authentication and authorization, you need a server-side component to verify and manage user access to resources. A dedicated backend can handle these tasks more effectively.
> 
> 3. **Scalability**: As your application grows, you may need to scale your SIWE implementation. A dedicated backend can provide the necessary infrastructure to scale your authentication and authorization processes.
> 
> 4. **Session Management**: Managing user sessions, including token-based authentication, can be challenging to handle on the client side alone. A backend server can help manage sessions and user states effectively.
> 
> To implement SIWE with Firebase or any other BaaS, you would typically follow a flow like this:
> 
> 1. The user initiates the SIWE process on the client side.
> 
> 2. The client sends a request to your dedicated backend, which interacts with the Ethereum blockchain.
> 
> 3. Your backend verifies the Ethereum signature to authenticate the user.
> 
> 4. Upon successful verification, your backend generates a Firebase Authentication token or other access tokens, which are sent back to the client.
> 
> 5. The client can then use these tokens to access Firebase services like Firestore, Storage, etc.
> 
> By using a dedicated backend in the flow, you can ensure that Ethereum-related operations are securely handled, and you can provide a consistent, scalable, and robust authentication system for your application. This approach offers better separation of concerns, security, and flexibility compared to attempting to handle everything on the client side.
> 
> ~ GPT, Chat

Given this, and in the case of Celebrity Fanalyzer, I would choose to proceed without a dedicated Backend.
Because I believe that it is not worth the effort/cost of creating a dedicated Backend knowing that we will soon migrate to Globe & Citizen.

**Comment by stravid87** [comment]
> 2\. The client sends a request to your dedicated backend, which interacts with the Ethereum blockchain.

"The client sends a request to your dedicated backend, which interacts with the Ethereum blockchain." Not true!

For now, I don't think we implement SIWE for C.F. However, I do suggest that we integrate Ethereum (more specifically the Polygon network) as our payment processor as soon as possible. Part of the SIWE exercise was to start getting used to concepts such as "provider", "signer", "private key", "public key" etc

**Comment by arnonrdp** [comment]
> For now, I don't think we implement SIWE for C.F. However, I do suggest that we integrate Ethereum (more specifically the Polygon network) as our payment processor as soon as possible. Part of the SIWE exercise was to start getting used to concepts such as "provider", "signer", "private key", "public key" etc

Having just the user's saved Ethereum address isn't enough to send payments to them?

**Comment by arnonrdp** [comment]
I'm closing this PR.
We have decided not to implement SIWE at this time.

**Pull request closed without merging**